### ISSUE TITLE
OpenSSL Engine PKCS11 created with succed but not listed

### ISSUE THREAD

**Issue by PaulSka** [openssl/openssl#17615]
Hello,
I upgrade OpenSSL 1.1.1l to OpenSSL 3.0.1 with PKCS11 dynamic engine (i don't have any problems with OpenSSL 1.1.1l)
I (think) correctly load PKCS11 (see above) but when i list engine, i have only the 2 default engine (rdrand and dynamic)

Version of two OpenSSL
```
OpenSSL 3.0.1 14 Dec 2021 (Library: OpenSSL 3.0.1 14 Dec 2021)
built on: Wed Dec 15 13:19:44 2021 UTC
platform: VC-WIN32
options:  bn(64,32)

OpenSSL 1.1.1l  24 Aug 2021
built on: Tue Aug 24 16:40:10 2021 UTC
platform: VC-WIN32
options:  bn(64,32) rc4(8x,mmx) des(long) idea(int) blowfish(ptr)
```

I use this to create my engine (same on OpenSSL 3.0.1) (CPS3_pkcs11 is a specific wrapper)
```
openssl engine dynamic -pre SO_PATH:C:\Windows\pkcs11.dll -pre ID:pkcs11 -pre LIST_ADD:1 -pre LOAD -pre MODULE_PATH:C:\Windows\SysWOW64\cps3_pkcs11_w32.dll
(dynamic) Dynamic engine loading support
[Success]: SO_PATH:C:\Windows\pkcs11.dll
[Success]: ID:pkcs11
[Success]: LIST_ADD:1
[Success]: LOAD
[Success]: MODULE_PATH:C:\Windows\SysWOW64\cps3_pkcs11_w32.dll
Loaded: (pkcs11) pkcs11 engine
```

Output to list engine
```
OpenSSL 3.0.1
openssl engine
(rdrand) Intel RDRAND engine
(dynamic) Dynamic engine loading support

OpenSSL 1.1.1l
openssl engine
(rdrand) Intel RDRAND engine
(dynamic) Dynamic engine loading support
(pkcs11) pkcs11 engine
```

I don't understand why my dynamic engine is loaded but not listed.

Thanks for your help

**Comment by petrovr** [comment]
Mathieu LYSAKOWSKI wrote:
> Hello,
> I upgrade OpenSSL 1.1X to OpenSSL 3.0 with PKCS11 dynamic engine (i don't have any problems with OpenSSL 1.1X)
X is important. If I remember well more recent 1.1.1 releases has correction for silly defect that "prevents" double loading of engine.
Remark: silly because it is user mistake not realistic issue.

Regards,
Roumen Petrov

**Comment by PaulSka** [comment]
I updated original post with full version of each OpenSSL
OpenSSL 1.1.1l and OpenSSL 3.0.1

**Comment by petrovr** [comment]
Mathieu LYSAKOWSKI wrote:
> I updated original post with full version of each OpenSSL
> OpenSSL 1.1.1l and OpenSSL 3.0.1
> 

And if 1.1.1m does not work look like regression from commits "Avoid loading of a dynamic engine twice".

Roumen

**Comment by nhorman** [comment]
Marking as inactive, to be closed at the end of the 3.4 dev cycle, barring further input
